tests/countries/test_turks_and_caicos_islands.py

```diff
@@ -0,0 +1,76 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.turks_and_caicos_islands import TurksAndCaicosIslands, TC, TCA
+from tests.common import CommonCountryTests
+
+
+class TestTC(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(TurksAndCaicosIslands, years=range(2020, 2025))
+
+    def test_country_aliases(self):
+        self.assertAliases(TurksAndCaicosIslands, TC, TCA)
+
```

Comment:
Please adds:
- test_no_checks

Movable holiday ones (see [`test_indonesia.py`](https://github.com/vacanza/holidays/blob/ce30a785023835e2f2e37d65944cc5aef0b64eec/tests/countries/test_indonesia.py#L243) for example)
- test_commonwealth_day
- test_good_friday
- test_easter_monday
- test_national_youth_day
- test_national_heritage_day
- test_national_day_of_thanksgiving

Fixed holidays with changed requirements that needs testing:
- test_jags_mccartney_day (name change; see [`test_macau.py`](https://github.com/vacanza/holidays/blob/ce30a785023835e2f2e37d65944cc5aef0b64eec/tests/countries/test_macau.py#L477) for reference)
- test_sovereign_birthday (for both king's and queen's birthday)